### ISSUE TITLE
Shage of class_weight in FocalLoss changed after saving checkpoint.

### ISSUE THREAD

**Issue by Hsuxu** [Project-MONAI/MONAI#7207]
When criterion and net are bundled, the shape `class_weight` in focal loss is changed in state_dict in after saving checkpoint. [line184-185](https://github.com/Project-MONAI/MONAI/blob/782c1e616312612c9f25bf4a33986ace67f7851e/monai/losses/focal_loss.py#L184C25-L184C25) change to `loss = self.class_weight.view(broadcast_dims) * loss` is a better way to avoid shape changing.

**Comment by wyli** [comment]
thanks for reporting, the `.view` is a light weight function without data copying, do you see   impact of speed with the current code? we may also consider inplace operation to save some space `loss *= self.class_weight.view(broadcast_dims)`.
